build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -120,7 +120,7 @@ public void setNumberOfNodes(int numberOfNodes) {
             throw new IllegalArgumentException("Number of nodes should be >= 1 but was " + numberOfNodes + " for " + this);
         }
 
-        if (numberOfNodes <= nodes.size()) {
+        if ((numberOfNodes != 1 && nodes.size() != 1) && numberOfNodes <= nodes.size()) {
```

Comment:
this is here to allow advertising `numberOfNodes = 1` in the default configuration

Reply:
I don't follow what is meant by "advertising" here. Isn't a cluster of 1 node the default. Why would it even be necessary to explicitly specify a size of `1`, and even so, what would be the problem with that w/o this change?

Reply:
> Isn't a cluster of 1 node the default.
yes
> Why would it even be necessary to explicitly specify a size of 1
It is not necessary, but allows me to define `numberOfNodes = 1` in the task definition to "advertise" you can have a more than one node by changing this value.  It is purely cosmetic and can remove this and the explicit definition of  `numberOfNodes = 1`

Reply:
I think we could just fix this by changing the conditional to `numberOfNodes < nodes.size()`. I'm not sure why we would fail when we set the number of nodes to the existing size anyway. We just want to protect against folks inadvertently trying to shrink a cluster.

Reply:
done [`dbda577` (#89563)](https://github.com/elastic/elasticsearch/pull/89563/commits/dbda577de2b43a4feb465b5397bcebe19cb103e5)